### DISCUSSION TITLE
VSD with GIFs

### DISCUSSION THREAD

**Discussion by su-aac** [UNC-Project-Open-AAC/OS-DPI#180] [Ideas]
Hello team, 

I want to share (1) the VSD with GIFs and (2) Core with GIFs that I made. 

(1) VSD with GIFs  
[VSD_with_GIFs_Sophie.zip](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/11240965/VSD_with_GIFs_Sophie.zip)
Here is the screenshot.
<img width="479" alt="image" src="https://user-images.githubusercontent.com/123387291/232260031-90097747-93f2-40ab-aaab-23aa68414e18.png">

(2) Core with GIFs
I don't know why I couldn't export the Core with GIFs file. The Export button under File is unclickable.  
Here is the screenshot.
<img width="484" alt="image" src="https://user-images.githubusercontent.com/123387291/232260039-eefbd7d7-c273-46a5-b752-733a68f7206b.png">

Sophie

**Comment by gbishop** [comment]
This is great! Edit not being available must be a bug. Try refreshing the browser, you shouldn't lose anything.

**Reply by gbishop** [reply]
It could be much smaller (I think) using mpeg 4 video instead of animated gifs. If animation is something we'd like to do it would be straightforward to extend to include mpeg 4. They are also nice because you can start and stop them. Maybe animate them on hover on when highlighted?

**Reply by gbishop** [reply]
Yes, a quick test shows that the mp4 versions are a factor of 4 to 40 smaller than the gifs.

**Comment by gbishop** [comment]
Were you ever able to export it? I hope I have fixed that issue in today's release.

**Comment by su-aac** [comment]
Thank you, Gary! I still cannot export it. I think the reason may be the gif files are too big. I'll try using mp4 and see if it fixes the problem.
Animating them on hover when highlighted is a brilliant idea! I would love to know how to program it.

Sophie

________________________________

**Comment by gbishop** [comment]
We don't support mp4 yet. Don't go down that road...

Try refreshing your browser and see if you can export.

**Comment by su-aac** [comment]
I tried, and it didn't work.

**Comment by gbishop** [comment]
https://fabricdigital.co.nz/blog/how-to-hard-refresh-your-browser-and-clear-cache#:~:text=Chrome%20and%20Mac%3A,click%20the%20'R'%20key
.

**Comment by gbishop** [comment]
If that doesn't work let's have a zoom meeting so I can investigate.

**Comment by su-aac** [comment]
That will be great! I can do Thursday and Friday after 10 AM.

**Comment by gbishop** [comment]
I have sent an invitation.

**Reply by su-aac** [reply]
Thank you!

**Reply by su-aac** [reply]
Hi Gary!
I just tried it, and it works!!!! I can export the file now. Thank you!!!

**Reply by su-aac** [reply]
The size of the file is too big to upload. 25 MB is allowed, and my file is 66 MB...

**Comment by gbishop** [comment]
Wow! That is big! I bet some of your gifs are really large.

**Reply by su-aac** [reply]
Yes.... I'll try to compress the gifs!

**Reply by su-aac** [reply]
Thank you for the help!

**Comment by gbishop** [comment]
For images in buttons, you only need about 100x100. If any are larger than
that, scale them down and replace them in the design.

**Comment by sofia-bg** [comment]
@gbishop been following this thread and no idea if this is possible, but any way that OS-DPI could auto-compress images or would that potentially be frustrating if it somehow reduces quality of image? Or maybe if file starting to get really big maybe a pop-up suggestion that the designer consider compressing images otherwise could have problem offloading?

**Comment by sofia-bg** [comment]
also random and maybe no one would want it but seeing @hsiaotingsu use pictures not symbols wondering if we could link in search engine like in tar heel reader? Or would that make the system unnecessarily complicated? Could also be useful for VSDs? 

Oh... or if we could add a search function for [Open Symbols](https://www.opensymbols.org/) to bring in symbols from that? Again no idea feasibility and obviously would need to also be able to name the symbol with the corresponding label you gave in spreadsheet (or be able to export the associated names as the images/symbols import??) 

Maybe not useful ideas but just stream of thought reflections.

**Comment by gbishop** [comment]
Please upload it to Google Drive. I'd like to experiment with making it
smaller.

**Reply by su-aac** [reply]
Hi Gary,
https://drive.google.com/file/d/1JVwgqkDAxnqPhvgz9J7XU2TlM51aw24I/view?usp=sharing
Here is the link to Google Drive. Thank you!

**Comment by gbishop** [comment]
I have extended the app to include mp4 videos as another type of image you can use. It expects the filename suffix to be exactly ".mp4". We could include other video formats if needed.

In your spreadsheet you can specify options for the video playback by including any of the following words space separated after the name in the symbol column.

- autoplay
- loop
- muted

The default is to play once when the button is cued.

I will attach the shrunken versions of Sophie's designs.
[Core_with_MP4s_Sophie.zip](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/11296277/Core_with_MP4s_Sophie.zip)
[VSD_with_MP4s_Sophie.zip](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/11296279/VSD_with_MP4s_Sophie.zip)

**Reply by su-aac** [reply]
This is awesome, Gary! I like that the default of playing once when the button is cued.
I downloaded the 2 zip files, but I need help with the next steps to see them display.

**Reply by aetkie** [reply]
[adding this note for documentation purposes] 
In order to see the designs that @gbishop linked above, download the zip file. Then change the ".zip" to ".osdpi". Then, you are able to Import the file in the OS-DPI.

**Reply by aetkie** [reply]
[another note for documentation purposes]

You can download the spreadsheet associated with a design from the OS-DPI. Here's how:

1. With the design open, click the File tab.
2. Save sheet

**Reply by gbishop** [reply]
You don't have to change the name. You can import it with the .zip suffix.

**Comment by gbishop** [comment]
Force your browser to refresh then import them

**Reply by su-aac** [reply]
Yeah, success! Thank you!

**Comment by gbishop** [comment]
I found a bug in my first version and have pushed an update. To prevent bad advertisers from showing you videos without permission the browser throws an exception if we try to play a video on hover before you have "interacted with" the site. Clicking anywhere appears to be sufficient to appease. it.

**Reply by gbishop** [reply]
I found a workaround. If you add the muted option to the video it will play without requiring a first click.
 
For example:

like.mp4 loop muted
